extension/httpfs/include/httpfs.hpp

```diff
@@ -6,37 +6,59 @@
 
 namespace duckdb_httplib_openssl {
 struct Response;
-}
-
+class Client;
+} // namespace duckdb_httplib_openssl
 namespace duckdb {
 
 using HeaderMap = unordered_map<string, string>;
 
-struct ResponseWrapper { /* avoid including httplib in header */
+// avoid including httplib in header
+struct ClientDeleter {
```

Comment:
A better way of doing this is to define the destructor of the object holding the unique_ptr in the `.cpp` file. That way this custom deleter is not required and the code can be simplified.